src/v/kafka/server/tests/create_topics_test.cc

```diff
@@ -447,27 +448,36 @@ FIXTURE_TEST(unlicensed_rejected, create_topic_fixture) {
       pandaproxy::schema_registry::schema_id_validation_mode::compat);
 
     revoke_license();
-    auto si_props = {
-      ss::sstring{kafka::topic_property_remote_read},
-      ss::sstring{kafka::topic_property_remote_write},
-      ss::sstring{kafka::topic_property_recovery},
-      ss::sstring{kafka::topic_property_read_replica},
-      ss::sstring{kafka::topic_property_record_key_schema_id_validation},
-      ss::sstring{kafka::topic_property_record_key_schema_id_validation_compat},
-      ss::sstring{kafka::topic_property_record_value_schema_id_validation},
-      ss::sstring{
-        kafka::topic_property_record_value_schema_id_validation_compat},
+    using prop_t = std::map<ss::sstring, ss::sstring>;
+    const auto with = [](const std::string_view prop, const auto value) {
+        return std::make_pair(
+          prop, prop_t{{ss::sstring{prop}, ssx::sformat("{}", value)}});
     };
+    std::vector<std::pair<std::string_view, prop_t>> enterprise_props{
+      // si_props
+      with(kafka::topic_property_remote_read, true),
+      with(kafka::topic_property_remote_write, true),
+      with(kafka::topic_property_recovery, true),
+      with(kafka::topic_property_read_replica, true),
+      // schema id validation
+      with(kafka::topic_property_record_key_schema_id_validation, true),
+      with(kafka::topic_property_record_key_schema_id_validation_compat, true),
+      with(kafka::topic_property_record_value_schema_id_validation, true),
+      with(
+        kafka::topic_property_record_value_schema_id_validation_compat, true),
```

Comment:
nitpick: It would be nice to split the refactor and the change into two commits.